pkg/mount-manager/safe-mounter_windows.go

```diff
@@ -0,0 +1,231 @@
+	if err != nil {
+		return nil, err
+	}
+	diskClient, err := diskclientv1alpha1.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	volumeClient, err := volumeclientv1alpha1.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	return &CSIProxyMounter{
+		FsClient:     fsClient,
+		DiskClient:   diskClient,
+		VolumeClient: volumeClient,
+	}, nil
+}
+
+func NewSafeMounter() (*mount.SafeFormatAndMount, error) {
+	csiProxyMounter, err := NewCSIProxyMounter()
+	if err != nil {
+		return nil, err
+	}
+	return &mount.SafeFormatAndMount{
+		Interface: csiProxyMounter,
+		Exec:      utilexec.New(),
+	}, nil
+}
+
+// Mount just creates a soft link at target pointing to source.
+func (mounter *CSIProxyMounter) Mount(source string, target string, fstype string, options []string) error {
+	// Mount is called after the format is done.
+	// TODO: Confirm that fstype is empty.
+	// Call the LinkPath CSI proxy from the source path to the target path
+	parentDir := filepath.Dir(target)
+	if err := os.MkdirAll(parentDir, 0755); err != nil {
+		return err
+	}
+	linkRequest := &fsv1alpha1.LinkPathRequest{
+		SourcePath: mount.NormalizeWindowsPath(source),
+		TargetPath: mount.NormalizeWindowsPath(target),
+	}
+	response, err := mounter.FsClient.LinkPath(context.Background(), linkRequest)
+	if err != nil {
+		return err
+	}
+	if response.Error != "" {
+		return errors.New(response.Error)
+	}
+	return nil
+}
+
+// Delete the given directory with Pod context. CSI proxy does a check for path prefix
+// based on context
+func (mounter *CSIProxyMounter) RemovePodDir(target string) error {
+	rmdirRequest := &fsv1alpha1.RmdirRequest{
+		Path:    mount.NormalizeWindowsPath(target),
+		Context: fsv1alpha1.PathContext_POD,
+		Force:   true,
+	}
+	_, err := mounter.FsClient.Rmdir(context.Background(), rmdirRequest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Delete the given directory with plugin context. CSI proxy does a check for path prefix
+// based on context
+func (mounter *CSIProxyMounter) RemovePluginDir(target string) error {
+	rmdirRequest := &fsv1alpha1.RmdirRequest{
+		Path:    mount.NormalizeWindowsPath(target),
+		Context: fsv1alpha1.PathContext_PLUGIN,
+		Force:   true,
+	}
+	_, err := mounter.FsClient.Rmdir(context.Background(), rmdirRequest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (mounter *CSIProxyMounter) Unmount(target string) error {
+	return mounter.RemovePodDir(target)
+}
+
+func (mounter *CSIProxyMounter) GetDevicePath(deviceName string, partition string, volumeKey string) (string, error) {
+	getDiskNumberRequest := &diskv1alpha1.GetDiskNumberByNameRequest{
+		DiskName: deviceName,
+	}
+	getDiskNumberResponse, err := mounter.DiskClient.GetDiskNumberByName(context.Background(), getDiskNumberRequest)
+	if err != nil {
+		return "", err
+	}
+	return getDiskNumberResponse.DiskNumber, nil
+
+}
+
+// formatAndMount - accepts the source disk number, target path to mount, the fstype to format with and options to be used.
```

Comment:
Could you mention that it mounts to the host?

Reply:
fixed